zio-config/src/main/scala/zio/config/ConfigSource.scala

```diff
@@ -2,30 +2,115 @@ package zio.config
 
 import zio.config.ReadErrors.ReadError
 import zio.system.System
-import zio.{ IO, ZIO }
+import zio.{ IO, UIO, ZIO }
 
-trait ConfigSource[K, V] {
+trait ConfigSource[K, V] { self =>
   val configSourceService: ConfigSource.Service[K, V]
+
+  final def orElse(that: ConfigSource[K, V]): ConfigSource[K, V] =
```

Comment:
I haven't looked at the implementation, but the operation makes perfect sense and is well-defined.

In addition, you could consider `<+>` / `orElseEither`, that allows the value types to vary (`V`, `V1`), returning `ConfigSource[K, Either[V, V1]]`.

Reply:
When we say orElse, are we considering the error that happens  in forming the configsource or when fetching the value for a vector of Paths?

Reply:
The latter, I think, since if you have `UIO[ConfigSource[K, V]]` (a "builder" for `ConfigSource`) then you can already `orElse` that together using ZIO `orElse`, with another "builder".

Reply:
Makes sense :)